run-tests.pl

```diff
@@ -448,21 +448,37 @@ sub fixture
    # actually invoke $setup
    @req_futures or push @req_futures, $f_start;
 
+   my $start = sub {
+      return if $f_start->is_ready;   # already started
+      $OUTPUT->diag( "Starting fixture $name" ) if ( $VERBOSE >= 2 );
+      $f_start->done( @_ );
+   };
+
    return Fixture(
       $name,
 
       \@requires,
 
-      sub { $f_start->done( @_ ) unless $f_start->is_ready },
+      $start,
 
       Future->needs_all( map { without_cancel($_) } @req_futures )
-         ->then( $setup )
-         ->set_label( $name ),
+         ->on_fail( sub {
+            my ( $reason ) = @_;
+
+            warn( "$name: input fixture failed with $reason" )
+               if ( $VERBOSE >= 3 );
+         })->then( $setup )->on_done( sub {
```

Comment:
Nit: I think we prefer to have one of these per line, to make it easier to follow the chain of whats going on

Reply:
done, though unconvinced it's much clearer